File: 프로그래머스/2/42842. 카펫/카펫.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int brown, int yellow) {
    vector<int> answer;
    for(int h = 1; h <= yellow / h; h++){
        if(yellow % h) continue;
        int bh = h + 2, bw = yellow / h + 2;
        if(bh * bw - yellow == brown){
            answer.push_back(bw);
            answer.push_back(bh);
            break;
        }
    }
    return answer;
}

File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        int yw = yellow, yh = 1, sum = brown + yellow;
        
        for(int i = 1; i <= sum; i++) {
            if(yellow % i == 0) {
                yw = yellow / i;
                yh = i;
                if(yh > yw) break;
                if((yw + 2) * (yh + 2) == sum) {
                    answer[0] = yw + 2;
                    answer[1] = yh + 2;
                }
            }
        } 
        
        return answer;
    }
}